src/main/java/com/grafana/loki/LokiSourceTask.java

```diff
@@ -0,0 +1,142 @@
+package com.grafana.loki;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.apache.hc.client5.http.classic.methods.HttpGet;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.ParseException;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.net.URIBuilder;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.source.SourceTask;
+import org.apache.kafka.connect.source.SourceTaskContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LokiSourceTask extends SourceTask {
+
+  private static final Logger log = LoggerFactory.getLogger(LokiSourceTask.class);
+
+  public static final String TIMESTAMP_FIELD = "timestamp";
+
+  private String lokiEndpoint;
+  private String lokiQuery;
+  private String topic;
+  private Long lastTimestamp = 0L;
+
+  private String authHeader;
+
+  @Override
+  public String version() {
+    return new LokiSourceConnector().version();
+  }
+
+  @Override
+  public void start(Map<String, String> props) {
+    lokiEndpoint = props.get(LokiSourceConnector.ENDPOINT_CONFIG);
+    lokiQuery = props.get(LokiSourceConnector.QUERY_CONFIG);
+    topic = props.get(LokiSourceConnector.TOPIC_CONFIG);
+
+    if (props.containsKey(LokiSourceConnector.START_CONFIG)) {
+      lastTimestamp = Long.parseLong(props.get(LokiSourceConnector.START_CONFIG));
+    } else {
+      lastTimestamp = now() - ONE_HOUR;
+    }
+
+    final String user = props.get(LokiSourceConnector.USERNAME_CONFIG);
+    final String password = props.get(LokiSourceConnector.PASSWORD_CONFIG);
+
+    authHeader = "Basic " + Base64.getEncoder().encodeToString((user + ":" + password).getBytes());
+  }
+
+  @Override
+  public void initialize(SourceTaskContext context) {
+    super.initialize(context);
+    lastTimestamp = 0L;
+    context.offsetStorageReader().offset(Collections.emptyMap());
+  }
+
+  @Override
+  public synchronized void stop() {}
+
+  @Override
+  public List<SourceRecord> poll() {
+    try (CloseableHttpClient client = HttpClients.createDefault()) {
+      final Long start = lastTimestamp + 1L;
+      final URI uri =
+          new URIBuilder(lokiEndpoint + "/loki/api/v1/query_range")
+              .addParameter("query", lokiQuery)
+              .addParameter("start", start.toString())
+              .addParameter("end", now().toString())
+              .addParameter("direction", "forward")
+              .build();
+
+      HttpGet request = new HttpGet(uri);
+      request.setHeader("Authorization", authHeader);
```

Comment:
I've tried for half an hour to use a credentials provider. I didn't succeed. This works however.